### ISSUE TITLE
Match Postgres Return Results

### ISSUE THREAD

**Issue by apavlo** [cmu-db/peloton#711]
The server does not return the correct status codes and results when executing queries from the terminal. We should always match what Postgres does.

For example, `CREATE` always returns zero regardless of whether the command succeeds or not. In this example I create the table twice. The first one is successful (because I can query against it) but returns zero. The second one presumably fails but it also returns zero without an error message.
```
postgres=# CREATE TABLE foo (id INT PRIMARY KEY);
CREATE TABLE foo (id INT PRIMARY KEY) 0
postgres=# SELECT * FROM foo;
 id 
----
(0 rows)

postgres=# CREATE TABLE foo (id INT PRIMARY KEY);
CREATE TABLE foo (id INT PRIMARY KEY) 0
```

Contrast this with **Postgres** output for the same commands:
```
pavlo=# CREATE TABLE foo (id INT PRIMARY KEY);
CREATE TABLE
pavlo=# SELECT * FROM foo;
 id 
----
(0 rows)

pavlo=# CREATE TABLE foo (id INT PRIMARY KEY);
ERROR:  relation "foo" already exists
```

It looks like we always just return the entire query string back to the user:
```
postgres=# INSERT INTO foo VALUES (123);
INSERT INTO foo VALUES (123) 1
```

Here's the same query in **Postgres**. I don't know what the "0" in the output that comes before the "1" means.
```
pavlo=# INSERT INTO foo VALUES (123);
INSERT 0 1
```

For `UPDATE` queries it looks like we are correctly returning the number of tuples modified:
```
postgres=# INSERT INTO foo VALUES (456);
INSERT INTO foo VALUES (456) 1
postgres=# UPDATE foo SET id = id + 1 WHERE id > 99;
UPDATE foo SET id = id + 1 WHERE id > 99 2
```

-----------------------

There are three things we need to do here:

1. Fix the return output of `INSERT`, `UPDATE`, and `DELETE` queries to only contain the string value of `QueryType` and not the entire SQL string.
2. Find out what the extra "0" in the return output of `INSERT`queries means in Postgres and make sure that we follow the same protocol.
3. Fix the output of `CREATE` to return the correct status code or error message based on whether the command succeeds.

For tasks (1) and (2) in the above list, I think that the only thing that you need to change is `PacketManager::CompleteCommand()`

https://github.com/cmu-db/peloton/blob/master/src/wire/packet_manager.cpp#L278

For fixing `CREATE` in (3), you should coordinate with the catalog team (@AngLi-Leon, @mengranwo, @camellyx) to find out why there is no exception thrown when you try to create a duplicate table. You may need to change the logic in `PacketManager::ExecQueryMessage()` to make sure this error messages gets propagated properly.

https://github.com/cmu-db/peloton/blob/master/src/wire/packet_manager.cpp#L342

**Comment by apavlo** [comment]
@grkamat will work on this issue.

**Comment by mengranwo** [comment]
We do check whether there is duplicate table already exist when creating a new one. 
However, we don't throw an exception, but instead set the transaction result type as **_failure_** and return. 
https://github.com/cmu-db/peloton/blob/master/src/catalog/catalog.cpp#L218

**Comment by apavlo** [comment]
@mengranwo We need to propagate the error message back to the client. You might need to throw an exception.

**Comment by apavlo** [comment]
@grkamat Make sure that you update this file to match the new output once you fix this issue. I re-enabled the psql tests:

https://github.com/cmu-db/peloton/blob/master/script/testing/psql/psql_ref_out.txt

**Comment by grkamat** [comment]
@apavlo 
The extra 0 in the return output of INSERT query are the Object Identifiers(OIDs) of the row created in postgres , in newer version of postgres, tables are created without OIDs on the rows that's why it gives 0.
I don't know where/how we are dealing with OIDs in peloton, will look into it.

**Comment by apavlo** [comment]
@grkamat We don't have OIDs for tuples yet. We will discuss this tomorrow at the team meeting.

**Comment by mengranwo** [comment]
@apavlo That is an easy fix, we can throw an exception when creating duplicate table or creating table with duplicate column names. But it seems that after throwing an exception and setting the transaction result type as failure, there is no proper response outside executor(i.e abort transaction immediately).

**Comment by dengyuchenkit** [comment]
Yes right now we return the response to the queries in Simple Query Protocol as the query itself (https://github.com/cmu-db/peloton/blob/master/src/wire/packet_manager.cpp#L278)
We can change the first input from query to the string value of the query type, if we want to change the response.
The '0' in the response to INSERT queries is now manually added.  (https://github.com/cmu-db/peloton/blob/master/src/wire/packet_manager.cpp#L278)
By looking at the output above, I found there's a mistake in #694  I submitted. One statement is missed in the code. 
The input to the MapToQueryType should be the string value of the query type, not the query itself. (https://github.com/cmu-db/peloton/blob/master/src/wire/packet_manager.cpp#L359) I should add one statement to parse out the string token of the query type before calling MapToQuery(). It's a quick fix. Sorry, should I submit a PR for it or it can be done in this PR?

**Comment by grkamat** [comment]
As per our discussion I used ParseQueryType and made some minor changes in the call to CompleteCommand() function but it seems like I will have to add some more cases(for CREATE and/or INSERT) because the output is not quite coherent with PostGres yet.

Peleton would produce something like this:
postgres=# CREATE TABLE foo (id INT PRIMARY KEY);
CREATE 0
mydb3=# CREATE TABLE foo (id INT PRIMARY KEY);
CREATE TABLE
postgres=# INSERT INTO foo VALUES (456);
INSERT 1
mydb3=# INSERT INTO foo VALUES (456);
INSERT 0 1
INSERT query is not hitting the case. Even though it is hard coded.
https://github.com/cmu-db/peloton/blob/master/src/wire/packet_manager.cpp#L293-L294 
Even on insert the default part of case is being hit
postgres=# UPDATE foo SET id = id + 1 WHERE id > 99;
UPDATE 1
mydb3=# UPDATE foo SET id = id + 1 WHERE id > 99;
UPDATE 1

**Comment by dengyuchenkit** [comment]
Do you change the ExecQueryMessage, here: https://github.com/cmu-db/peloton/blob/master/src/wire/packet_manager.cpp#L359 (as in the previous comment)? The first param to the MapToQueryType should be the string value of the query type not the query itself. Right now it's not giving the correct query type

**Comment by grkamat** [comment]
I changed that and it gives the correct output for INSERT. Thank you.
CREATE status code remains a concern.

**Comment by apavlo** [comment]
Fixed in #734
